commit db1b9b53be979b11bbf0c7235ea4446a70930e22
Author: Albert Astals Cid <aacid@kde.org>
Date:   Sun Nov 9 23:01:58 2014 +0100

    Don't report gphoto devices that are child of gphoto devices
    
    Acked by Lukáš
    
    REVIEW: 119834

diff --git a/solid/solid/backends/udev/udevmanager.cpp b/solid/solid/backends/udev/udevmanager.cpp
index 1f6369f..a8393c0 100644
--- a/solid/solid/backends/udev/udevmanager.cpp
+++ b/solid/solid/backends/udev/udevmanager.cpp
@@ -130,7 +130,7 @@ bool UDevManager::Private::checkOfInterest(const UdevQt::Device &device)
            device.subsystem() == QLatin1String("video4linux") ||
            device.subsystem() == QLatin1String("net") ||
            device.deviceProperty("ID_MEDIA_PLAYER").toString().isEmpty() == false || // media-player-info recognized devices
-           device.deviceProperty("ID_GPHOTO2").toInt() == 1; // GPhoto2 cameras
+           (device.deviceProperty("ID_GPHOTO2").toInt() == 1 && device.parent().deviceProperty("ID_GPHOTO2").toInt() != 1); // GPhoto2 cameras
 }
 
 bool UDevManager::Private::isLidBubtton(const UdevQt::Device& device)
